prowler/providers/gcp/gcp_provider.py

```diff
@@ -35,33 +37,58 @@ class GcpProvider(Provider):
     def __init__(self, arguments):
         logger.info("Instantiating GCP Provider ...")
         input_project_ids = arguments.project_id
+        excluded_project_ids = arguments.excluded_project_id
         credentials_file = arguments.credentials_file
+        list_project_ids = arguments.list_project_id
 
         self._session = self.setup_session(credentials_file)
 
         self._project_ids = []
         self._projects = {}
+        self._excluded_project_ids = []
         accessible_projects = self.get_projects()
         if not accessible_projects:
             logger.critical("No Project IDs can be accessed via Google Credentials.")
             sys.exit(1)
 
         if input_project_ids:
             for input_project in input_project_ids:
-                if input_project in accessible_projects.keys():
-                    self._projects[input_project] = accessible_projects[input_project]
-                    self._project_ids.append(accessible_projects[input_project].id)
-                else:
-                    logger.critical(
-                        f"Project {input_project} cannot be accessed via Google Credentials."
-                    )
-                    sys.exit(1)
+                for accessible_project in accessible_projects:
+                    if self.is_project_matching(input_project, accessible_project):
+                        self._projects[accessible_project] = accessible_projects[
+                            accessible_project
+                        ]
+                        self._project_ids.append(
+                            accessible_projects[accessible_project].id
+                        )
         else:
             # If not projects were input, all accessible projects are scanned by default
             for project_id, project in accessible_projects.items():
                 self._projects[project_id] = project
                 self._project_ids.append(project_id)
 
+        # Remove excluded projects if any input
+        if excluded_project_ids:
+            for excluded_project in excluded_project_ids:
+                for project_id in self._project_ids:
+                    if self.is_project_matching(excluded_project, project_id):
+                        self._excluded_project_ids.append(project_id)
```

Comment:
Why don't you do the following three lines at line 75? I mean if you know that the project wants to be excluded you don't need to iterate again through all the `self._excluded_project_ids`, you can just do the `pop` and `remove` right?

Reply:
No, because the excluded_project could have an asterisk and apply to one or more project ids.

Reply:
Right, thanks for clarifying this.